nixos/modules/services/hardware/vdr.nix

```diff
@@ -0,0 +1,99 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.vdr;
+
+  libDir = "/var/lib/vdr";
+in {
+
+  ###### interface
+
+  options = {
+
+    services.vdr = {
+      enable = mkEnableOption "VDR";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.vdr-with-plugins;
+        description = "Package to use.";
+      };
+
+      shutdown = mkOption {
+        type = types.path;
+        default = pkgs.writeScript "vdr-shutdown" ''
+          #!${pkgs.stdenv.shell} -eu
+          next="$2"
+          if [ "$next" -eq 0 ]; then # no timer
+            next=86400 # one day
+          elif [ "$next" -lt 0 ]; then # recording is running
+            next=60 # one minute
+          fi
+          /run/wrappers/bin/sudo ${pkgs.utillinux}/bin/rtcwake -m off -s "$next"
+          '';
+        description = "Shutdown command";
+      };
+
+      videoDir = mkOption {
+        type = types.path;
+        default = "/srv/vdr/video";
+        description = "Recording directory";
+      };
+
+      extraArguments = mkOption {
+        type = types.str;
```

Comment:
Better to make this a list.

Reply:
right, much better. Will use it with lib.escapeShellArgs